Upstream i386 ULPs are generated on a machine with SSE2 support, which
gives slightly more precision as the SSE2 cosf and sinf functions 
internally uses double precisions registers.

On the other hand the x86 FPU has higher internal precision than the
SSE2 one for some other functions.

Finally for the complex arc cosine functions family, the expected
results are wrongly rounded according to match the results from Intel
CPUs, why AMD CPU are actually more precise by 1 ULP.

This patch therefore changes the ULPs to match the lower precisions of
the 3 above cases, given the actual function used is decided at runtime
depending on the CPU capabilities.

--- a/sysdeps/i386/fpu/libm-test-ulps
+++ b/sysdeps/i386/fpu/libm-test-ulps
@@ -192,9 +192,9 @@
 ldouble: 2
 
 Function: Real part of "cacos_downward":
-double: 1
+double: 2
 float: 1
-idouble: 1
+idouble: 2
 ifloat: 1
 ildouble: 2
 ldouble: 2
@@ -208,9 +208,9 @@
 ldouble: 5
 
 Function: Real part of "cacos_towardzero":
-double: 1
+double: 2
 float: 1
-idouble: 1
+idouble: 2
 ifloat: 1
 ildouble: 2
 ldouble: 2
@@ -264,9 +264,9 @@
 ldouble: 5
 
 Function: Imaginary part of "cacosh_downward":
-double: 1
+double: 2
 float: 1
-idouble: 1
+idouble: 2
 ifloat: 1
 ildouble: 2
 ldouble: 2
@@ -280,9 +280,9 @@
 ldouble: 5
 
 Function: Imaginary part of "cacosh_towardzero":
-double: 1
+double: 2
 float: 1
-idouble: 1
+idouble: 2
 ifloat: 1
 ildouble: 2
 ldouble: 2
@@ -916,24 +916,32 @@
 ldouble: 1
 
 Function: "cos":
+float: 1
+ifloat: 1
 ildouble: 1
 ldouble: 1
 
 Function: "cos_downward":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 3
 ldouble: 3
 
 Function: "cos_towardzero":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 2
 ldouble: 2
 
 Function: "cos_upward":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 2
 ldouble: 2
 
@@ -1243,17 +1251,17 @@
 
 Function: Real part of "ctan_towardzero":
 double: 3
-float: 1
+float: 2
 idouble: 3
-ifloat: 1
+ifloat: 2
 ildouble: 5
 ldouble: 5
 
 Function: Imaginary part of "ctan_towardzero":
 double: 2
-float: 1
+float: 3
 idouble: 2
-ifloat: 1
+ifloat: 3
 ildouble: 4
 ldouble: 4
 
@@ -1307,17 +1315,17 @@
 
 Function: Real part of "ctanh_towardzero":
 double: 2
-float: 1
+float: 3
 idouble: 2
-ifloat: 1
+ifloat: 3
 ildouble: 4
 ldouble: 4
 
 Function: Imaginary part of "ctanh_towardzero":
 double: 2
-float: 1
+float: 2
 idouble: 2
-ifloat: 1
+ifloat: 2
 ildouble: 3
 ldouble: 3
 
@@ -1816,46 +1824,62 @@
 ldouble: 2
 
 Function: "sin":
+float: 1
+ifloat: 1
 ildouble: 1
 ldouble: 1
 
 Function: "sin_downward":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 2
 ldouble: 2
 
 Function: "sin_towardzero":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 1
 ldouble: 1
 
 Function: "sin_upward":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 3
 ldouble: 3
 
 Function: "sincos":
+float: 1
+ifloat: 1
 ildouble: 1
 ldouble: 1
 
 Function: "sincos_downward":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 3
 ldouble: 3
 
 Function: "sincos_towardzero":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 2
 ldouble: 2
 
 Function: "sincos_upward":
 double: 1
+float: 1
 idouble: 1
+ifloat: 1
 ildouble: 3
 ldouble: 3
 
